operatorcourier/validate.py

```diff
@@ -237,3 +243,231 @@ def _type_validation(self, bundleData, typeName, validator, required):
+                        "required" : True
+                    },
+                    {
+                        "field" : "links",
+                        "description" : "Without this field, no links will be displayed in the details page side panel.",
+                        "required" : True
+                    },
+                    {
+                        "field" : "icon",
+                        "description" : "Without this field, the operator will display a default operator framework icon.",
+                        "required" : False
+                    },
+                    {
+                        "field" : "provider",
+                        "description" : "Without this field, users will not be able to filter for provider.",
+                        "required" : True
+                    },
+                    {
+                        "field" : "maintainer",
+                        "description" : "Without this field, the operator details page will not display the name and contact for users to get support in using the operator.",
+                        "required" : True
+                    },
+                    {
+                        "field" : "customresourcedefinitions",
+                        "description" : "Without this field, the operator details page will not display any CRDs the operator needs in order to function.",
+                        "required" : False
+                    },
+                    {
+                        "field" : "installModes",
+                        "description" : "Without this field, OLM will not be able to deploy your operator.",
+                        "required" : True
+                    }
+                ]
+
+                for field in spec_required_fields:
+                    if field["field"] not in csv["spec"]:
+                        if field["required"]:
+                            logger.error("csv spec.%s not defined. %s", field["field"], field["description"])
+                            valid = False
+                        else:
+                            logger.warning("csv spec.%s not defined. %s", field["field"], field["description"])
+
+        return valid
+    
+    def _ui_csv_fields_format_validation(self, csv):
+
+        def is_url(field):
+            if not v.url(field):
+                return False
+            return True
+
+        def is_email(field):
+            if not v.email(field):
+                return False
+            return True
+
+        def is_version(field):
+            pattern = re.compile(r'v(\d+\.)(\d+\.)(\d)')
+            return pattern.match(field)
+
+        def is_capability_level(field):
+            levels = [
+                "Basic Install",
+                "Seamless Upgrades",
+                "Full Lifecycle",
+                "Deep Insight",
+                "Auto Pilot"
+            ]
+            return field in levels
+
+        def is_date(field):
```

Comment:
What's the expected format for date? @jeff-phillips-18 is there a convention we need for our db?